compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2399,10 +2399,10 @@ impl DerefMut for Target {
 
 impl Target {
     /// Given a function ABI, turn it into the correct ABI for this target.
-    pub fn adjust_abi(&self, abi: Abi) -> Abi {
+    pub fn adjust_abi(&self, abi: Abi, c_variadic: bool) -> Abi {
         match abi {
             Abi::C { .. } => self.default_adjusted_cabi.unwrap_or(abi),
-            Abi::System { unwind } if self.is_like_windows && self.arch == "x86" => {
+            Abi::System { unwind } if self.is_like_windows && self.arch == "x86" && !c_variadic => {
```

Comment:
Can you add a link to some reference that states authoritatively that on x86, "system" should mean "C" with varargs but "stdcall" without varargs?